src/main/java/net/shadowmage/ancientwarfare/structure/tile/TileDecorativeFlag.java

```diff
@@ -0,0 +1,65 @@
+package net.shadowmage.ancientwarfare.structure.tile;
+
+import net.minecraft.item.ItemStack;
+import net.minecraft.nbt.NBTTagCompound;
+import net.shadowmage.ancientwarfare.structure.init.AWStructureBlocks;
+
+public class TileDecorativeFlag extends TileFlag {
+
+	@Override
+	protected void handleUpdateNBT(NBTTagCompound tag) {
+		super.handleUpdateNBT(tag);
+		readNBT(tag);
+	}
+
+	private void readNBT(NBTTagCompound tag) {
+		topColor = tag.getInteger("topColor");
+		bottomColor = tag.getInteger("bottomColor");
+		name = tag.getString("name");
+	}
+
+	@Override
+	public void readFromNBT(NBTTagCompound compound) {
+		super.readFromNBT(compound);
+		readNBT(compound);
+		name = compound.getString(NAME_TAG);
+	}
+
+	@Override
+	protected void writeUpdateNBT(NBTTagCompound tag) {
+		super.writeUpdateNBT(tag);
+		writeNBT(tag);
+	}
+
+	@Override
+	public NBTTagCompound writeToNBT(NBTTagCompound compound) {
+		NBTTagCompound tag = writeNBT(super.writeToNBT(compound));
+		tag.setString(NAME_TAG, name);
```

Comment:
serializing name should be done in writeNBT as that's used both for initial load of tile as well as for serializing when this tile is synced to client. This way the name wouldn't sync there.